x-pack/plugin/core/src/test/java/org/elasticsearch/index/engine/FrozenEngineTests.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.index.engine;
+
+import org.apache.lucene.search.MatchAllDocsQuery;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.store.AlreadyClosedException;
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.lucene.index.ElasticsearchDirectoryReader;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.core.internal.io.IOUtils;
+import org.elasticsearch.index.mapper.ParsedDocument;
+import org.elasticsearch.index.seqno.SequenceNumbers;
+import org.elasticsearch.index.store.Store;
+import org.elasticsearch.indices.breaker.HierarchyCircuitBreakerService;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class FrozenEngineTests extends EngineTestCase {
+
+    public void testAcquireReleaseReset() throws IOException {
+        IOUtils.close(engine, store);
+        final AtomicLong globalCheckpoint = new AtomicLong(SequenceNumbers.NO_OPS_PERFORMED);
+        try (Store store = createStore()) {
+            EngineConfig config = config(defaultSettings, store, createTempDir(), newMergePolicy(), null, null, globalCheckpoint::get);
+            try (InternalEngine engine = createEngine(config)) {
+                int numDocs = Math.min(10, addDocuments(globalCheckpoint, engine));
+                engine.flushAndClose();
```

Comment:
My first thought was that this isn't ok because the global checkpoint may be stuck below 10 and we won't have enough data in the safe commit. Upon better inspection, the read only index doesn't care and opens the last commit anyway, so all is good. That said, I rather leave the seq# universe totally out of this - can we maybe just make sure the global checkpoint is always the local checkpoint and that don't skip seq nos?